Solutions/STM32F4DISCOVERY/TinyBooter/TinyBooter.proj

```diff
@@ -23,6 +23,8 @@
         <ExtraTargets>$(ExtraTargets);CompressBin</ExtraTargets>
         <ScatterFileDefinition>scatterfile_bootloader_$(COMPILER_TOOL).$(SCATTER_EXT)</ScatterFileDefinition>
         <EXEScatterFileDefinition>scatterfile_bootloader_$(COMPILER_TOOL).$(SCATTER_EXT)</EXEScatterFileDefinition>
+        <!-- newlib-nano does not support long-long and long-double formatting, use standard newlib instead -->

```

Comment:
IIRC, the Tinybooter code does not require long-long and long-double formatting support, but the TinyCLR does (uses "%I64" and "%lld" printf formats). The newlib-nano included in GCC toolchain does not have those options enabled (i.e. long.ToString() and double.ToString() does not work correctly), so there are basically two options:

1. Use default newlib-nano in TinyBooter and standard newlib in TinyCLR,
2. Build a custom newlib-nano with long-long and long-double formatting support and use it in both TinyBooter and TinyCLR.

The ideal solution would be to use default newlib-nano in Tinybooter and a custom newlib-nano in TinyCLR - I guess that could be done with some additional tweaking of linker settings, but it would probably be too confusing...

Unfortunately, I have not yet had a chance to review the recent GCC versions, sometimes they introduce changes in newlib-nano, so if they for example added long-long and long-double formatting by default, it could made things a little bit easier.